src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -81,6 +81,20 @@ public static bool IsUnnecessaryCast(CastExpressionSyntax cast, SemanticModel se
             if (IsRemovableBitwiseEnumNegation(cast, semanticModel, cancellationToken))
                 return true;
 
+            // Special case for converting a method group to object. The compiler issues a warning if the cast is removed:
+            // warning CS8974: Converting method group 'ToString' to non-delegate type 'object'. Did you intend to invoke the method?
+            var castExpressionOperation = semanticModel.GetOperation(cast.Expression, cancellationToken);
+            if (castExpressionOperation?.Kind == OperationKind.MethodReference &&
+                semanticModel.GetTypeInfo(cast.Type, cancellationToken).Type?.SpecialType is SpecialType.System_Object)
+            {
+                // If we have a double cast, report as unnecessary, e.g:
+                // (object)(object)MethodGroup
+                // (Delegate)(object)MethodGroup
+                // If we have a single object cast, don't report as unnecessary e.g:
+                // (object)MethodGroup
+                return castExpressionOperation.Parent?.Parent?.Kind == OperationKind.Conversion;
```

Comment:
Please add test where it is cast to object, then down to T, where T is constrained to a delegate type.